Templates/GitlabCIPipeline/README.md

```diff
@@ -0,0 +1,175 @@
+
+The pipeline uses the Gitlab concepts: `Stage`and `Jobs`.
+
+![Gitlab Release Pipeline](images/gitlab-pipeline-release.png)
+
+## Prerequisites
+
+To leverages this template, access to a Gitlab CI environment is required, and an Gitlab runner must be configured to connect to your mainframe environment. Please review the setup instructions of this [document](https://www.ibm.com/support/pages/system/files/inline-files/Integrating%20IBM%20zOS%20platform%20in%20CICD%20pipelines%20with%20GitLab%20-%20v1.7_1.pdf).
+
+The template leverages the [Zowe CLI](https://docs.zowe.org/stable/user-guide/cli-installcli/) to issue command invoking the [Common Backend scripts](../Common-Backend-Scripts/) on USS.
+* Zowe `base` and `rse` profiles needs to be configured in the `zowe.config.json` file under `.zowe` directory
+
+    Example of Zowe configuration file:
+    ```
+    "$schema": "./zowe.schema.json",
+    "profiles": {
+        "rse": {
+            "type": "rse",
+            "properties": {
+                "port": 6800,
+                "basePath": "rseapi",
+                "protocol": "https"
+            },
+            "secure": []
+        },
+        "base": {
+            "type": "base",
+            "properties": {
+                "host": "mainframe.hostname",
+                "rejectUnauthorized": false
+            },
+            "secure": []
+        }
+    },
+    "defaults": {
+        "rse": "rse",
+        "base": "base"
+    },
+    "autoStore": false
+    ```
+
+The [Common Backend scripts](../Common-Backend-Scripts/) need to be configured for the selected deployment technologies to operate correctly.
+
+## Installation and setup of template
+
+**Note: Please work with your pipeline specialist to review the below section.**
+
+The `.gitlab-ci.yaml` can be dropped into the root folder of your Gitlab Git repository and will automatically provide pipelines for the specified triggers. Please review the definitions thoroughly with your Gitlab administrator.
+
+Following requirements need to be met:
+* Zowe rse profile is configured to connect to z/OS machine as described above.
+
+### Variables configuration
+The following variables need to be defined and configured as the environment variables in the GitLab group or project setting:
+
+Variables | Description
+--- | ---
+AutomationToken | [Group access token](https://docs.gitlab.com/ee/api/rest/#personalprojectgroup-access-tokens) to be used for authentication when invoke Gitlab CLI REST interfaces.
+RSEAPI_USER | Username for Zowe RSEAPI server authentication. This username is used when issue shell script command through Zowe.
+RSEAPI_PASSWORD | Password for Zowe RSEAPI server authentication. This password is used when issue shell script command through Zowe.
+PIPELINE_WORKSPACE | Root directory on z/OS Unix System services to perform builds. E.g. /u/gitlab/workspace
+WAZI_DEPLOY_CONFIGDIR | Path to a directory of Wazi Deploy configuration files.
```

Comment:
I think we should say this `WAZI_DEPLOY_CONFIGDIR` directory is on USS and can be populated with the samples available in https://github.com/jbyibm/cics-genapp/tree/main/wazideploy-samples